src/gballoc.c

```diff
@@ -168,12 +169,12 @@ void* gballoc_calloc(size_t nmemb, size_t size)
             {
                 /* Codes_SRS_GBALLOC_01_021: [If the underlying calloc call is successful, gballoc_calloc shall increment the total memory used with nmemb*size.] */
                 allocation->ptr = result;
-                allocation->size = nmemb * size;
+                allocation->size = safe_multiply_size_t(nmemb, size);
                 allocation->next = head;
                 head = allocation;
                 g_allocations++;
 
-                totalSize += allocation->size;
+                totalSize = safe_add_size_t(totalSize, allocation->size);
```

Comment:
Should check for overflow.

Reply:
not needed. This is only for malloc stats update